base-isa.md

```diff
@@ -22,48 +23,45 @@ The highest two bits of the first byte are a format marker:
-| D      | displacement                 |
-| S      | size (res.d)                 |
-| ?      | reserved for unknown purpose |
+| Symbol | Meaning         |
+|--------|-----------------|
+| C      | Opcode          |
+| R      | Register Id     |
+| I      | Immediate       |
+| D      | Displacement    |
+| S      | Operation Size  |
+| M      | Memory Mode     |
+| ?      | Arbitrary Value |
 
-### Execution of Reserved Instructions
-
-Execution of a reserved instruction is undefined behavior. A future extension will change this to "the CPU must trigger a specific interrupt."
+### Execution of Illegal and Reserved Instructions
 
+Execution of an _illegal_ or _reserved_ instruction is _unspecified_ behavior. A future extension will change this to require an interrupt to occur.
 
 ## Computation Instructions
 
-Both computation formats share a lot of similarities. For both the first byte has the structure `0x SS CCCC`.
+Both computation formats share a lot of similarities. For both, the first byte has the structure `0x SS CCCC`.
 
-- `SS` is a size marker and reserved for extensions. For the base instructions this should always be `01`
-- `CCCC` is a 4bit opcode deciding which operation to execute. This can be an ALU or a memory load/store instruction
+- `SS` is a size marker and reserved for extensions. For the base instructions this _must_ always be `01`
+- `CCCC` is a 4 bit opcode deciding which operation to execute.
 
 ### 2 Register Computation
 
-The second byte has the format `AAA BBB 00` where `AAA` and `BBB` are references to registers. `AAA` is the destination register and left source register. `BBB` is the right source register. The only exception is the RSUB instruction where `AAA` and `BBB` are swapped for the purpose of being source registers.
+The second byte has the format `AAA BBB MM` where `AAA` and `BBB` are references to registers. `AAA` is the destination register and left source register. `BBB` is the right source register. There are 2 exceptions to this. The first is the RSUB instruction where `AAA` and `BBB` are swapped for the purpose of being source registers. The second exception is that the `B` argument specifies which memory address or control register to write to. `MM` is the memory mode which will be defined in an extension and is only defined for `00` in the base isa.
 
 ### Immediate Computation
 
-The second byte has the format `RRR IIIII`, where the 5bit immediate acts as operand `B`. The immediate is sign extended for operations 0-7 and operation 9. The immediate is zero extended for operation 8 and operations 10-15. While interleaving of signed and unsigned immediate at the border of the 2 sections seems unusual at first glance, it's done to make decoding hardware simpler.
+The second byte has the format `RRR IIIII`, where the 5 bit immediate acts as operand `B`. The immediate is sign extended for operations 0-7 and operation 9. The immediate is zero extended for operation 8 and operations 10-15.
```

Comment:
Since we've recently established that operations 8 and 9 are equally easy to detect, I'm completely in favor of swapping `movs` and `movz` back to the way they used to be. This is a breaking change but simplifies documentation.

Also `movs` is probably more common in practice so my brain likes the idea of it having opcode `8` because that number is "nicer." My brain is weird, and I have no idea how much this is biasing my favor for swapping those opcodes back :P

Reply:
I'm against changing it because it doesn't affect decoding in any significant way AND is a breaking change.

Reply:
If we decide to change how `test` works, we can IMO also at the same time swap movs and movz.

Reply:
I remembered my reasoning for having movz at 8 and movs at 9. It's because you can use the 1s bit directly in an AND gate without needing to negate it first. So it saves a single gate.

Reply:
I don't think breaking is significant since it's indistinguishable in base and there is 1 single nontrivial ETCa program out there written for a relevant extension. That said, I agree with Mega; it can stay for now.